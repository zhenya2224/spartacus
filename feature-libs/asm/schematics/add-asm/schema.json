{
  "$schema": "http://json-schema.org/schema",
  "$id": "AsmSchematics",
  "title": "ASM Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the asm features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["ASM"]
    }
  },
  "required": []
}
