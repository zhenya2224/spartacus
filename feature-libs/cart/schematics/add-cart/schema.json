{
  "$schema": "http://json-schema.org/schema",
  "$id": "CartSchematics",
  "title": "Cart Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Saved cart feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Saved-Cart", "Quick-Order", "Import-Export"],
        "type": "string"
      },
      "default": ["Saved-Cart", "Quick-Order", "Import-Export"],
      "x-prompt": {
        "message": "Which Cart features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Saved-Cart",
            "label": "Saved Cart"
          },
          {
            "value": "Quick-Order",
            "label": "Quick Order"
          },
          {
            "value": "Import-Export",
            "label": "Import/Export"
          }
        ]
      }
    }
  },
  "required": []
}
