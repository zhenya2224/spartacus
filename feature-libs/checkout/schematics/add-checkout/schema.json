{
  "$schema": "http://json-schema.org/schema",
  "$id": "CheckoutSchematics",
  "title": "Checkout Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Checkout feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Checkout"]
    }
  },
  "required": []
}
