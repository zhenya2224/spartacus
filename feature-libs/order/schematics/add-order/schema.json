{
  "$schema": "http://json-schema.org/schema",
  "$id": "OrderSchematics",
  "title": "Order Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Order feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Order"]
    }
  },
  "required": []
}
