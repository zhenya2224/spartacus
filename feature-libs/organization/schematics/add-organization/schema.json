{
  "$schema": "http://json-schema.org/schema",
  "$id": "OrganizationSchematics",
  "title": "Organization Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the organization features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Administration", "Order-Approval"],
        "type": "string"
      },
      "default": ["Administration", "Order-Approval"],
      "x-prompt": {
        "message": "Which Organization features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Administration",
            "label": "Administration (b2b feature)"
          },
          { "value": "Order-Approval", "label": "Order Approval (b2b feature)" }
        ]
      }
    }
  },
  "required": []
}
