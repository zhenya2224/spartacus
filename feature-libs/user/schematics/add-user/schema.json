{
  "$schema": "http://json-schema.org/schema",
  "$id": "UserSchematics",
  "title": "User Schematics",
  "description": "Most of the Spartacus features require User Account feature to be properly configured.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the user features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Account", "Profile"],
        "type": "string"
      },
      "default": ["Account", "Profile"],
      "x-prompt": "Which features would you like to set up from the User library? Please note that for most Spartacus features to be properly configured, the Account feature is required."
    }
  },
  "required": []
}
