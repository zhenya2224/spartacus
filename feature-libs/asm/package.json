{
  "name": "@spartacus/asm",
  "version": "4.1.0-next.0",
  "description": "ASM feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "asm"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/asm",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/forms": "^12.0.5",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
