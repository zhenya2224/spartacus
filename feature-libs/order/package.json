{
  "name": "@spartacus/order",
  "version": "4.2.0",
  "description": "Order feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "order"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/order",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^12.0.5",
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/forms": "^12.0.5",
    "@angular/router": "^12.0.5",
    "@ng-bootstrap/ng-bootstrap": "^10.0.0",
    "@ng-select/ng-select": "^7.0.1",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "@spartacus/styles": "4.1.0-next.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
