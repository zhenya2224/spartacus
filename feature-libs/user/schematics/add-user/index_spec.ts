/// <reference types="jest" />

import { RunSchematicTaskOptions } from '@angular-devkit/schematics/tasks/run-schematic/options';
import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CLI_USER_ACCOUNT_FEATURE,
  CLI_USER_PROFILE_FEATURE,
  LibraryOptions,
  LibraryOptions as SpartacusUserOptions,
  SpartacusOptions,
  SPARTACUS_SCHEMATICS,
  SPARTACUS_USER,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');
const featureModulePath =
  'src/app/spartacus/features/user/user-feature.module.ts';
const scssFilePath = 'src/styles/spartacus/user.scss';

describe('Spartacus User schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusUserOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const accountFeatureOptions: SpartacusUserOptions = {
    ...libraryNoFeaturesOptions,
    features: [CLI_USER_ACCOUNT_FEATURE],
  };

  const profileFeatureOptions: SpartacusUserOptions = {
    ...libraryNoFeaturesOptions,
    features: [CLI_USER_PROFILE_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        SPARTACUS_SCHEMATICS,
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', libraryNoFeaturesOptions, appTree)
        .toPromise();
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(featureModulePath)).toBeFalsy();
    });

    it('should install necessary Spartacus libraries', () => {
      const packageJson = JSON.parse(appTree.readContent('package.json'));
      let dependencies: Record<string, string> = {};
      dependencies = { ...packageJson.dependencies };
      dependencies = { ...dependencies, ...packageJson.devDependencies };

      for (const toAdd in peerDependencies) {
        // skip the SPARTACUS_SCHEMATICS, as those are added only when running by the Angular CLI, and not in the testing environment
        if (
          !peerDependencies.hasOwnProperty(toAdd) ||
          toAdd === SPARTACUS_SCHEMATICS
        ) {
          continue;
        }
        // TODO: after 4.0: use this test, as we'll have synced versions between lib's and root package.json
        // const expectedVersion = (peerDependencies as Record<
        //   string,
        //   string
        // >)[toAdd];
        const expectedDependency = dependencies[toAdd];
        expect(expectedDependency).toBeTruthy();
        // expect(expectedDependency).toEqual(expectedVersion);
      }
    });
  });

  describe('Account feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', accountFeatureOptions, appTree)
          .toPromise();
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });

      describe('styling', () => {
        it('should create a proper scss file', () => {
          const scssContent = appTree.readContent(scssFilePath);
          expect(scssContent).toMatchSnapshot();
        });

        it('should update angular.json', async () => {
          const content = appTree.readContent('/angular.json');
          expect(content).toMatchSnapshot();
        });
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...accountFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });

  describe('Profile feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', profileFeatureOptions, appTree)
          .toPromise();
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });

      describe('styling', () => {
        it('should create a proper scss file', () => {
          const scssContent = appTree.readContent(scssFilePath);
          expect(scssContent).toMatchSnapshot();
        });

        it('should update angular.json', async () => {
          const content = appTree.readContent('/angular.json');
          expect(content).toMatchSnapshot();
        });
      });

      it('should run the proper installation tasks', async () => {
        const tasks = schematicRunner.tasks
          .filter((task) => task.name === 'run-schematic')
          .map(
            (task) => task.options as RunSchematicTaskOptions<LibraryOptions>
          );
        expect(tasks.length).toEqual(1);

        const userTaskWithSubFeatures = tasks[0];
        expect(userTaskWithSubFeatures).toBeTruthy();
        expect(userTaskWithSubFeatures.name).toEqual('add-spartacus-library');
        expect(userTaskWithSubFeatures.options).toHaveProperty(
          'collection',
          SPARTACUS_USER
        );
        expect(userTaskWithSubFeatures.options.options?.features).toEqual([
          CLI_USER_ACCOUNT_FEATURE,
        ]);
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...profileFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(featureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
