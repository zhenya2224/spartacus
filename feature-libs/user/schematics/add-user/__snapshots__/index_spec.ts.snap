// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus User schematics: ng-add Account feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { UserAccountModule } from \\"@spartacus/user/account\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule } from \\"@spartacus/user/account/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule,
    UserAccountModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  })]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Account feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { userAccountTranslationChunksConfig, userAccountTranslations } from \\"@spartacus/user/account/assets\\";
import { UserAccountRootModule, USER_ACCOUNT_FEATURE } from \\"@spartacus/user/account/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserAccountRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [USER_ACCOUNT_FEATURE]: {
        module: () =>
          import('@spartacus/user/account').then((m) => m.UserAccountModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userAccountTranslations,
      chunks: userAccountTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Account feature general setup styling should create a proper scss file 1`] = `"@import \\"@spartacus/user\\";"`;

exports[`Spartacus User schematics: ng-add Account feature general setup styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        },
        \\"@schematics/angular:application\\": {
          \\"strict\\": true
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"500kb\\",
                  \\"maximumError\\": \\"2.5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"2kb\\",
                  \\"maximumError\\": \\"4kb\\"
                }
              ],
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"outputHashing\\": \\"all\\"
            },
            \\"development\\": {
              \\"buildOptimizer\\": false,
              \\"optimization\\": false,
              \\"vendorChunk\\": true,
              \\"extractLicenses\\": false,
              \\"sourceMap\\": true,
              \\"namedChunks\\": true
            }
          },
          \\"defaultConfiguration\\": \\"production\\"
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            },
            \\"development\\": {
              \\"browserTarget\\": \\"schematics-test:build:development\\"
            }
          },
          \\"defaultConfiguration\\": \\"development\\"
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;

exports[`Spartacus User schematics: ng-add Profile feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { UserProfileModule } from \\"@spartacus/user/profile\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserProfileRootModule,
    UserProfileModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Profile feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from \\"@spartacus/core\\";
import { userProfileTranslationChunksConfig, userProfileTranslations } from \\"@spartacus/user/profile/assets\\";
import { UserProfileRootModule, USER_PROFILE_FEATURE } from \\"@spartacus/user/profile/root\\";

@NgModule({
  declarations: [],
  imports: [
    UserProfileRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [USER_PROFILE_FEATURE]: {
        module: () =>
          import('@spartacus/user/profile').then((m) => m.UserProfileModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: userProfileTranslations,
      chunks: userProfileTranslationChunksConfig,
    },
  })
  ]
})
export class UserFeatureModule { }
"
`;

exports[`Spartacus User schematics: ng-add Profile feature general setup styling should create a proper scss file 1`] = `"@import \\"@spartacus/user\\";"`;

exports[`Spartacus User schematics: ng-add Profile feature general setup styling should update angular.json 1`] = `
"{
  \\"$schema\\": \\"./node_modules/@angular/cli/lib/config/schema.json\\",
  \\"version\\": 1,
  \\"newProjectRoot\\": \\"\\",
  \\"projects\\": {
    \\"schematics-test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        },
        \\"@schematics/angular:application\\": {
          \\"strict\\": true
        }
      },
      \\"root\\": \\"\\",
      \\"sourceRoot\\": \\"src\\",
      \\"prefix\\": \\"app\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/schematics-test\\",
            \\"index\\": \\"src/index.html\\",
            \\"main\\": \\"src/main.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.app.json\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"500kb\\",
                  \\"maximumError\\": \\"2.5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"2kb\\",
                  \\"maximumError\\": \\"4kb\\"
                }
              ],
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"src/environments/environment.ts\\",
                  \\"with\\": \\"src/environments/environment.prod.ts\\"
                }
              ],
              \\"outputHashing\\": \\"all\\"
            },
            \\"development\\": {
              \\"buildOptimizer\\": false,
              \\"optimization\\": false,
              \\"vendorChunk\\": true,
              \\"extractLicenses\\": false,
              \\"sourceMap\\": true,
              \\"namedChunks\\": true
            }
          },
          \\"defaultConfiguration\\": \\"production\\"
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"schematics-test:build:production\\"
            },
            \\"development\\": {
              \\"browserTarget\\": \\"schematics-test:build:development\\"
            }
          },
          \\"defaultConfiguration\\": \\"development\\"
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"schematics-test:build\\"
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:karma\\",
          \\"options\\": {
            \\"main\\": \\"src/test.ts\\",
            \\"polyfills\\": \\"src/polyfills.ts\\",
            \\"tsConfig\\": \\"tsconfig.spec.json\\",
            \\"karmaConfig\\": \\"karma.conf.js\\",
            \\"inlineStyleLanguage\\": \\"scss\\",
            \\"assets\\": [
              \\"src/favicon.ico\\",
              \\"src/assets\\"
            ],
            \\"styles\\": [
              \\"src/styles.scss\\",
              \\"src/styles/spartacus/user.scss\\"
            ],
            \\"scripts\\": []
          }
        }
      }
    }
  },
  \\"defaultProject\\": \\"schematics-test\\"
}"
`;
